spatialmath/pose_test.go

```diff
@@ -172,3 +173,31 @@ func BenchmarkPoseBetween(b *testing.B) {
 		PoseBetween(p1b, p2b)
 	}
 }
+
+var result Pose
+
+func BenchmarkCompose(b *testing.B) {
+	r := rand.New(rand.NewSource(517))
+	var x Pose
+	for n := 0; n < b.N; n++ {
+		x = Compose(randPose(r), randPose(r))
+	}
+	// Prevent compiler from interfering with benchmark
+	result = x
+}
+
+// Use this to measure the actual speed of Compose above.
+func BenchmarkRandPose(b *testing.B) {
```

Comment:
Probably easier to make this all part of one function with the work being done to create the test cases done up front then writing a subtest that runs through the test cases.  That way simply running the benchmark reports the relevant info without needing to do extra math here on the benchmark results

Reply:
Good idea, will do

Reply:
Improved the benchmark, no need for math any longer.

Averaging the results of running the benchmark 5x on main and on this branch, I'm seeing a speedup per-iteration of the raw `Compose()` call from 165ns to 157ns. Which is about a 5% improvement.